# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/GlobalObject.cpp
# Commit: 410d1b57bb09
# Full Hash: 410d1b57bb0955e0fe2f47b95ffa5ee1005e6791
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-09 21:37:17
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 5: Replace heap state checks that rely on there being a JSContext for the thread r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140192
# ==============================================================================

diff -r f78d139e1d0e -r 410d1b57bb09 js/src/vm/GlobalObject.cpp
--- a/js/src/vm/GlobalObject.cpp	Wed Mar 09 10:17:07 2022 +0000
+++ b/js/src/vm/GlobalObject.cpp	Wed Mar 09 10:17:07 2022 +0000
@@ -1030,8 +1030,8 @@
 GlobalObjectData::GlobalObjectData(Zone* zone) : varNames(zone) {}
 
 void GlobalObjectData::trace(JSTracer* trc, GlobalObject* global) {
-  // Atoms are always tenured.
-  if (!JS::RuntimeHeapIsMinorCollecting()) {
+  // Atoms are always tenured so don't need to be traced during minor GC.
+  if (trc->runtime()->heapState() != JS::HeapState::MinorCollecting) {
     varNames.trace(trc);
   }
 